src/plugins/unified_search/public/query_string_input/query_bar_top_row.tsx

```diff
@@ -486,6 +508,39 @@ export const QueryBarTopRow = React.memo(
       return <EuiFlexItem className={wrapperClasses}>{component}</EuiFlexItem>;
     }
 
+    function renderCancelButton() {
+      const buttonLabelCancel = strings.getCancelQueryLabel();
+
+      if (submitButtonIconOnly) {
+        return (
+          <EuiButtonIcon
+            iconType="cross"
+            aria-label={buttonLabelCancel}
+            onClick={onClickCancelButton}
+            size={shouldShowDatePickerAsBadge() ? 's' : 'm'}
+            data-test-subj="queryCancelButton"
+            color="danger"
+            display="base"
+          >
+            {buttonLabelCancel}
+          </EuiButtonIcon>
+        );
+      }
+
+      return (
+        <EuiButton
+          iconType="cross"
+          aria-label={buttonLabelCancel}
+          onClick={onClickCancelButton}
+          size={shouldShowDatePickerAsBadge() ? 's' : 'm'}
+          data-test-subj="queryCancelButton"
+          color="danger"
```

Comment:
@ppisljar I think the only thing I don't like is this danger color. If we change this in text I think it looks better (and less intimidating)

Wdyt @davismcphee ? 

![change](https://github.com/elastic/kibana/assets/17003240/cda3f6d5-1e8e-4f36-ae25-7b5b2743e58b)

Other than that I think is a very nice addition!